fcrepo-http-commons/src/main/java/org/fcrepo/http/commons/domain/Range.java

```diff
@@ -135,6 +119,111 @@ public static Range convert(final String source) {
             end = parseLong(to);
         }
 
-        return new Range(start, end, no_start);
+        return new Range(start, end);
+    }
+
+    /**
+     * Create a range object with start and end bytes based on the length of the content.
+     * @param length the length of the content
+     * @return a range of length object
+     */
+    public Range.RangeOfLength rangeOfLength(final long length) {
+        final long end;
+
+        if (
+            end() == -1 || // Range end is not specified
+            end() >= length || // Range end is too large
+            start() == -1 // Range start is not specified
+        ) {
+            end = length - 1;
+        } else {
+            end = end();
+        }
+
+        final long start;
+        if (start() == -1) {
+            start = length - size();
+        } else {
+            start = start();
+        }
+        return new RangeOfLength(start, end, length);
+    }
+
+    /**
+     * Represents a range object based on length
+     */
+    public static class RangeOfLength {
+
+        /**
+         * The start of the range
+         */
+        private final long start;
+
+        /**
+         * The end of the range
+         */
+        private final long end;
+
+        /**
+         * Is the range satisfiable
+         */
+        private final boolean satisfiable;
+
+        /**
+         * Create a range object based on length
+         * @param start the start of the range
+         * @param end the end of the range
+         * @param length the length of the content
+         */
+        protected RangeOfLength(final long start, final long end, final long length) {
+            this.start = start;
+            this.end = end;
+            // If a valid byte-range-set includes at least one byte-range-spec with a first-byte-pos that is less than
+            // the current length of the representation, or at least one suffix-byte-range-spec with a non-zero
+            // suffix-length, then the byte-range-set is satisfiable. Otherwise, the byte-range-set is unsatisfiable.
+            this.satisfiable = start < length && (end() == -1 || end > start) && (start != -1 && end != -1);
```

Comment:
I think `end > start` should be `end >= start`. At the moment if I request `-H"Range: bytes=0-0"` I get a 416, but it should give me the first byte (even the comment down at line 223 agrees). We should probably add a case to RangeTest for this.

Reply:
I made that comment, so it could be wrong. I will make this change because I feel like that comment and you are correct.

Reply:
I wasn't really sure from reviewing the spec what the expected outcome was, but I tried it against github (which supports range headers) and `0-0` does get me the first byte there. Plus there wouldn't be a way to get a single byte otherwise, since the ranges are inclusive so`0-1` returns 2 bytes.